src/core/scalebar/qgsscalebarsettings.h

```diff
@@ -551,18 +602,12 @@ class CORE_EXPORT QgsScaleBarSettings
     //! Text format
     QgsTextFormat mTextFormat;
 
-    //! Fill color
-    QColor mFillColor = QColor( 0, 0, 0 );
-    //! Secondary fill color
-    QColor mFillColor2 = QColor( 255, 255, 255 );
-    //! Fill
-    QBrush mBrush;
-    //! Secondary fill
-    QBrush mBrush2;
     //! Height of bars/lines
     double mHeight = 3.0;
 
     std::unique_ptr< QgsLineSymbol > mLineSymbol;
+    std::unique_ptr< QgsFillSymbol > mFillSymbol;
```

Comment:
Depending on whether you want to implement the suggestion above, maybe adding a 1 at the end of the variable?